### PR TITLE
Query planner improvements

### PR DESCRIPTION
**Pull request by baitsguy**
1. Magical prompting and example improvements for:
1 Using vector search when appropriate
2 Use match_phrase for proper nouns
1. Disable vector-search-removal from default strategy

### CODE REVIEW

File: lib/sycamore/sycamore/query/operators/query_database.py [hunk 1]
```diff
@@ -32,12 +32,8 @@ class QueryDatabase(Node):
                             }
                         }
                     },
-                    {
-                        "match": {
-                            "properties.path.keyword": "/path/to/data/*.pdf",
-                            "properties.entity.location": "Georgia"
-                        }
-                    }
+                    {"match": {"properties.entity.accidentNumber.keyword": "*K1234N*"}},
```
Comment (mdwelsh) [inline lib/sycamore/sycamore/query/operators/query_database.py:35]: Y U NO FORMAT

Comment (mdwelsh) [reply lib/sycamore/sycamore/query/operators/query_database.py:35]: Also the example text above needs to be updated if you are including the K1234N thing in the query.

File: lib/sycamore/sycamore/query/operators/query_database.py [hunk 2]
```diff
@@ -63,19 +62,15 @@ def output_type(self) -> type:
...
-        and the final answer doesn't require you to scan the full dataset (e.g. find me an instance of a transcript
-        that talked about AI).
-    2. Use QueryVectorDatabase for RAG-style questions where you need to retrieve a set of candidate records where the
-        answer might exist.
-    3. Do not use QueryVectorDatabase when examining each record is essential (e.g. did all transcripts talk about AI?).
-    4. Do not use QueryVectorDatabase where you need to potentially return large datasets (e.g. give me all transcripts
-        that talked about AI).
 
+    You should only use QueryVectorDatabase for the following query types:
+    1. "Is there *any* <record> similar to <query>?" - yes or no questions about inclusion of any record
```
Comment (mdwelsh) [inline lib/sycamore/sycamore/query/operators/query_database.py:67]: I'm a stickler for ending sentences with periods on the (probably superstitious) belief that this helps the LLM tokenize better :-)

**Review by mdwelsh — APPROVED** [review]

File: lib/sycamore/sycamore/query/operators/query_database.py (hunk 1 above)
Comment (baitsguy) [reply lib/sycamore/sycamore/query/operators/query_database.py:35]: oh typo nice catch

**Pull request merged**